packages/sitecore-jss-react/src/components/BYOCComponent.tsx

```diff
@@ -6,14 +6,24 @@ import * as FEAAS from '@sitecore-feaas/clientside/react';
 
 export const BYOC_COMPONENT_RENDERING_NAME = 'BYOCComponent';
 
+/**
+ * FEaaS props for server rendering.
+ */
+type BYOCServerProps = {
+  /**
+   * Fetched data from server component props for server rendering, based on rendering params.
+   */
+  fetchedData?: FEAAS.DataScopes;
+};
+
 /**
  * Data from rendering params on Sitecore's BYOC rendering
  */
 export type BYOCComponentParams = {
   /**
    * Name of the component to render
    */
-  ComponentName: string;
+  ComponentName?: string;
```

Comment:
Is _ComponentName_ optional?

Reply:
We need to have all the props as optional to avoid type conflicts when reading props from LayoutData